Import for COCIRS_0xxx, COCIRS_1xxx, and VG_28xx

juzen2003 opened the pull request:
This pull request (base on https://github.com/SETI/pds-opus/pull/1174) is to track the progress of adding COCIRS_0xxx and COCIRS_1xxx to OPUS: (need to work with https://github.com/SETI/pds-webtools/pull/40)

Items done:
Add import steps for COCIRS_0xxx and COCIRS_1xxx 
To do items:
- In pdsfile, add rules for COCIRS_0xxx and COCIRS_1xxx
- In OPUS, update the ring geometry values
- In OPUS, add api results test for COCIRS_0xxx and COCIRS_1xxx

juzen2003 committed:
Include steps to import VG_2801
- In config_data.py, add VG_28xx related info to VOLSETS_WITH_PREVIEWS,
  INSTRUMENT_ABBREV_TO_MISSION_ABBREV, INSTRUMENT_ABBREV_TO_INSTRUMENT_NAME,
  VOLUME_ID_PREFIX_TO_INSTRUMENT_NAME, VOLUME_ID_PREFIX_TO_MISSION_ABBREV,
  VOLUME_ID_ROOT_TO_TYPE, and TARGET_NAME_MAPPING. Also Create
  VOLUME_ID_TO_INSTRUMENT_NAME to store Mapping from VOLUME ID to
  instrument name for VG_28xx.
- In do_import.py (line 171-172), modify create_tables_for_import to
  fetch the correct instrument from VOLUME_ID_PREFIX_TO_INSTRUMENT_NAME
  for VG_28xx.
- In do_import.py (line 712-716), modify import_one_index to fetch
  the correct instrument from VOLUME_ID_PREFIX_TO_INSTRUMENT_NAME
  for VG_28xx.
- In import_util.py, update yield_import_volume_ids by adding 'VG_28xx'.
- In populate_obs_instrument_VGISS.py (line 322-327), update
  populate_obs_mission_voyager_VGISS_mission_phase_name_OBS to fetch
  MISSION_PHASE_NAME from index file.
- In populate_obs_mission_voyager.py (line 85-86), update
  populate_obs_mission_voyager_ert to handle the case when ert is not
  available.
- In populate_obs_mission_voyager.py (line 102-145), update
  populate_obs_mission_voyager_spacecraft_clock_count1/2 to handle the
  case when SPACECRAFT_CLOCK_PARTITION_NUMBER is not available.
- In populate_obs_mission_voyager.py (line 50-57), update
  helper_voyager_planet_id to fetch planet id from TARGET_NAME for VG_28xx.
- In populate_util.py (line 146-150), update _star_name_helper to get the
  star name from SIGNAL_SOURCE_NAME_1 for VG_28xx.
- In populate_util.py (line 136-142), add populate_target_name_from_index
  to fetch target name from TARGET_NAME in index file.
- In obs_general.json (line 81-84), add VG_28xx instruments to mult options
  for instrument_id.
- In obs_mission_voyager.json (line 132-133), change the data source of
  mission_phase_name from index file to function because VG_28xx doesn't
  have mission_phase_name in index file.
- Create import/populate_obs_instrument_VGPPS_occ.py to import for VG_2801.

juzen2003 committed:
Add steps to import VG_2802 (VGUVS)
- Create populate_obs_instrument_VGUVS_occ.py based on
  populate_obs_instrument_VGPPS_occ.py.

juzen2003 committed:
Add steps to import VG_2803 (RSS)
- Update populate_obs_occultation_VGRSS/VGPPS/VGUVS_source_OCC to fetch
  source from SIGNAL_SOURCE_NAME_1 in index file.

juzen2003 committed:
Update the *_wl_band_OCC & *_source_OCC to fetch wavelength band from
both WAVELENGTH_BAND_1 & WAVELENGTH_BAND_2 and occultation source from
SIGNAL_SOURCE_NAME_1 & SIGNAL_SOURCE_NAME_2.

juzen2003 committed:
Add steps to import VG_2810
- In config_data.py (line 200,209), add VG_2810 related info to
  VOLUME_ID_TO_INSTRUMENT_NAME and VOLUME_ID_PREFIX_TO_INSTRUMENT_NAME.
- Create populate_obs_instrument_VGISS_occ.py to import VG_2810.
- In config_data.py (line 47), import populate_obs_instrument_VGISS_occ.
- In populate_obs_instrument_VGISS.py (line 334,343,353), add type to
  functions to distinguish between OBS and OCC.
- In obs_instrument_vgiss.json (line 102,130,131), add N/A as one of mult
  options for shutter_mode.
- In obs_occultation.json (line 275-282), add "Visual" as one of mult
  options for wl_band.

juzen2003 committed:
Fix the observation_type, quantity, and occ_type for VG_2810.
- In obs_occultation.json (line 79-83), add "Reflectance" as one of the
  occ_type stored in db.
- In populate_obs_instrument_VGISS_occ.py (line 202-203), return "REF"
  in populate_obs_occultation_VGISS_occ_type_OCC.
- In populate_obs_instrument_VGISS_occ.py (line 70), return "REFECT" in
  populate_obs_general_VGISS_quantity_OCC.
- In populate_obs_instrument_VGISS_occ.py (line 73-74), return "INT" in
  populate_obs_general_VGISS_observation_type_OCC.

juzen2003 committed:
Fixed import error by updating the logic in do_import.py to make sure
both EBROCC and VG_28xx instrument_name are properly handled.
- In do_import.py (line 174-179, 719), update the logic to properly
  handle instrument_name of both EBROCC and VG_28xx.
- In import_for_tests.sh, put back all volumes (including VG_28xx) for
  test import.
- In populate_obs_mission_voyager.py (line 107,109,135,137), properly
  fetch "VOLUME_ID" from index_label to determine if we should get
  spacecraft_clock_count1/2 from supplemental index or profile index files.

juzen2003 committed:
Rename all populate_obs_*_occ.py to populate_obs_*_prof.prof.py

juzen2003 committed:
Remove extra space at the return of populate_obs_instrument_*_channel_OCC

juzen2003 committed:
Remove unused functions in populate_obs_instrument_*_prof.py

juzen2003 committed:
Rename all functions in populate_*_prof.py, change the suffix from "OCC"
to "PROF". Also change "OCC" to "PROF" in VOLUME_ID_ROOT_TO_TYPE in
config_data.py.

juzen2003 committed:
Set the default widget to Instrument Name, Observation Type, and Intended
Target Name.
- In settings.py (line 308), update opus default widget to
'instrument,observationtype,target'.

juzen2003 committed:
Update all ring elevation calculations for VGISS_prof.

juzen2003 committed:
Update ring geometry calculations for VGRSS_PROF

juzen2003 committed:
Fix the calculations for ring geo in VGRSS_PROF

juzen2003 committed:
Update ring geometry calculations for VGUVS_PROF.

juzen2003 committed:
Add calculations of ring geometry in VGPPS_PROF.

juzen2003 committed:
Fixed and cleaned up variables naming in each populate_*_VG*_prof.py to
make sure import can be run without errors.

juzen2003 committed:
Remove the extra space at line 543 in populate_obs_instrument_VGRSS_prof.py

juzen2003 committed:
Update Django tests  results on default widgets, image intensity, tables
triggered by mission Voyager.

juzen2003 committed:
Add api results tests for VG_28xx, here are the list of opus ids:
- 'co-vims-occ-2014-175-l2pup-e',
- 'vg-pps-2-s-occ-1981-238-delsco-e',
- 'vg-pps-2-u-occ-1986-024-delta-sigsgr-e',
- 'vg-pps-2-n-occ-1989-236-sigsgr-i',
- 'vg-uvs-1-s-occ-1980-317-iother-e',
- 'vg-rss-2-u-occ-1986-024-four-s43-i',
- 'vg-uvs-2-n-occ-1989-236-sigsgr-i',
- 'vg-iss-2-s-prof'
(Will have to update tests when profile_index files are available)

juzen2003 committed:
Update the return of populate_obs_occultation_*_source_PROF and
populate_obs_occultation_*_wl_band_PROF. Now they will return a string.

juzen2003 committed:
Fixed yield_import_volume_ids by adding options for VGPPS/VGUVS/VGRSS.
- In test__results.py (line 426-460), add triggered table tests on VGPPS
  , VGUVS, and VGRSS. (will have to fix this later)
- In config_data.py (line 203), remove unused dictionary.
- In import_util.py (line 108-124), add options options for VGPPS/VGUVS/
  VGRSS in yield_import_volume_ids.

juzen2003 committed:
Fixed the mults options for wl_band in obs_occultation.json and observation_type
in obs_general.json.
- In obs_general.json (line 571), change the "intensity Profile" to
  "Reflectance".
- In obs_occultation.json(line 282), remove "VISUAL", use "VISIBLE".
- In populate_obs_instrument_VGISS_prof.py (line 255-256), return "VI"
  (visible) for the wavelength band.

juzen2003 committed:
- Removed commented out codes in populate_obs_general_*_inst_host_id_PROF.
- Removed unused function _integration_duration_helper.
- Updated comments in populate_obs_ring_geometry_*_incidence1_PROF.
- Renamed populate_obs_instrument_VGISS.py to populate_obs_instrument_VGISS_obs.py

juzen2003 committed:
Clean up code styles in populate_obs_instrument_*_prof.py.

juzen2003 committed:
Create a new function _is_voyager_at_north_except_uranus to pass into
functions that calculate ring elevations.
- In populate_obs_instrument_VGPPS_prof.py (line 371-378) and
  populate_obs_instrument_VGUVS_prof.py (line 375-382), create
  _is_voyager_at_north_except_uranus to calculate ring elevations.

juzen2003 committed:
Rename "obs_occultation" table to "obs_profile"

juzen2003 committed:
Rename obs_occultation.json to obs_profile.json.

juzen2003 committed:
Renamed "Occultation Constraints" to "Occultation&Reflectance Profile Constraints".

juzen2003 committed:
- Renamed "Occultation Constraints" to "Ring/Limb Profile Constraints".
- Change the observation type of VG_2810 to "Reflectance Profile".

juzen2003 committed:
Update test results for VG_28xx.
- In create_test_results.py (line 70-76), update opus_id for VG_28xx due
  to the change of rules in pdsfile.
- Create new test results based on new opus ids for VG_28xx.

juzen2003 committed:
Update based on github feedback.
- Change "Ring/Limb Profile" to "Occultation and Reflectance Profiles".
- Add tooltips for new opus_type in internal_def_product_types.json.

juzen2003 committed:
Chang "Occultation and reflectance Profiles" to "Occultation/Reflectance Profiles".

juzen2003 committed:
Add tooltip for vgrss_occ_inv05 & vgrss_occ_inv0_4 and re-organize the
order of product type in internal_def_product_types.json.

juzen2003 committed:
Update the wording of product types tooltips in internal_def_product_types.json.

juzen2003 committed:
Merge branch 'import_for_VG_28xx_occ' into import_COCIRS_0_1xxx

juzen2003 committed:
Update import steps to be able to import COCIRS_0xxx and COCIRS_1xxx.
- In config_data.py (line 235-236), add COCIRS_0xxx and COCIRS_1xxx to
  VOLUME_ID_ROOT_TO_TYPE.
- In import_util.py (line 38-39, 66-67, 90-91, 103-104), add COCIRS_0xxx
  and COCIRS_1xxxyield_import_volume_ids.
- In populate_obs_instrument_COCIRS.py (line 346-353), add
  obs_instrument_coiss_detector_id_OBS to obtain detector id for different
  COCIRS volumes.
- In populate_obs_instrument_COCIRS.py (line 25-33),add _get_COCIRS_file_spec
  to get the file spec for different COCIRS volumes.
- In populate_obs_instrument_COCIRS.py, update the way we fetch file spec,
  mission phase name, and detector in COCIRS.
- In obs_instrument_cocirs.json (line 48-49), update the use the function
  obs_instrument_coiss_detector_id_OBS to fetch the detector id for
  different COCIRS volumes.

juzen2003 committed:
Update the way to fetch wavenumber for COCIRS_0xxx and COCIRS_1xxx in
populate_obs_instrument_COCIRS.py,
- In populate_obs_instrument_COCIRS.py (line 369-389), create
  _get_COCIRS_max_waveno and _get_COCIRS_min_waveno to fetch wavenumber
  for different volumes.

juzen2003 committed:
Update the way to fetch spec size for different volumes of COCIRS.
- In populate_obs_instrument_COCIRS.py (line 229-236), update how we
  fetch spec size for different volumes.

juzen2003 committed:
Change field type field of opus id in table schema. Change from char40
to char50 to accommodate longer opus id. (Might have to change it back
once we finalize the opus id for COCIRS_0xxx/1xxx)

juzen2003 committed:
Fetch wavenumber resolution from BAND_BIN_WIDTH in supplemental index for
COCIRS_0xxx/1xxx.
- In populate_obs_instrument_COCIRS.py (line 186-187, 171-172), fetch wavenumber
  resolution from BAND_BIN_WIDTH in supplemental_index for COCIRS_0xxx/1xxx.

juzen2003 committed:
Add populate functions for ring geo fields in
populate_obs_instrument_COCIRS.py (line 127-331)

juzen2003 committed:
Update to import surface geo using *cube_equi_index*
- In config_data.py (line 1129-1132), add INDEX_FILES_WITH_SURFACE_GEO_INFO,
  this is used to store partial index files basenames that contain surface
  geometry info.
- In do_import.py (line 799-813), check if the index file contains surface
  geo info, if so, we set the body_surface_geo to sub_assoc_type and
  import the surface geo info later.
- In do_import.py (line 929-951), construct surface geo dict for index
  files that contains surface geo info. This will be used to import
  surface geo data later.
- In obs_surface_geometry_target.json, update data source to import data
  using populate functions when surface geo data doesn't exist in index
  files.
- In populate_obs_instrument_COCIRS.py (line 340-543), add populate funcitons
  to import surface geo from equi index files.

juzen2003 committed:
Clean up the issues raised by flake8.
- In populate_obs_instrument_COCIRS.py (line 136,146,156,166,184,194), make
  sure populate functions have proper return.
- In populate_obs_instrument_COCIRS.py (line 249,260,296,306), change
  one line if statments to multiple lines.

juzen2003 committed:
Add result tests for COCIRS_0xxx/1xxx.

juzen2003 committed:
Update results tests for COCIRS_0xxx/1xxx.

juzen2003 committed:
Add tooltips for opus_type: cocirs_cube, cocirs_extra, cube_index.
- In internal_def_product_types.json (line 74-78, 344-353), add tooltips
  for cocirs_cube, cocirs_extra, and cube_index.

juzen2003 committed:
Add 'GOOD' as the data quality score for VG_28xx.
- Update all populate_obs_occultation_*_quality_score_PROF for VG_28xx
  to return 'GOOD'.

juzen2003 committed:
- Update right_asc and declination values for BET_PER & IOT_HER in STAR_RA_DEC
in config_data.py.
- Update the spec_flag to return yes for VGUVS.

juzen2003 committed:
Update VG ISS constraints for VGISS_PROF. Now the info for those constraints
can also be fetched from populate functions.
- In populate_obs_instrument_VGISS_prof.py (line 525-550), add populate
  functions to generate data for VGISS constraints.
- In obs_instrument_vgiss.json, update data source to fetch VGISS constraints
  from populate functions if they don't exists in the index files.

juzen2003 committed:
Update image constraints data for VGISS_PROF.
- In populate_obs_instrument_VGISS_prof.py (line 110-124), update return
  values of populate functions for image constraints.

juzen2003 committed:
Updated some VG_28xx import data based on Mark's spreadsheet
- Update data quality score to 'GOOD' for VG_28xx
- Update right_asc and declination values for the following stars in STAR_RA_DEC:
    'BET_PER':              ( 47.042218542,  40.955646667)
    'IO_HER':               (264.866193333,  46.006331945)
- Change the spec size to None for VGUVS (since it's TBD)
- Add data for VGISS constraints and image constraints for VGISS (prof)
Note: for ring geometry like north-based incidence angle/emission angle,
the populate functions still return the results of mathematic calculations
instead of None (they are TBD in the spread sheet)

juzen2003 committed:
Update the results tests.

juzen2003 committed:
Update populate_obs_general_COCIRS_observation_type_OBS to return spectral
cube as the observation type if 'CUBE' exists in file spec name.

juzen2003 committed:
- Update the return observation type for COCIRS_0xxx/1xxx populate function.
- Update results tests.

rfrenchseti commented on opus/application/apps/help/templates/help/gettingstarted.html:
```diff
@@ -42,7 +42,7 @@ <h3>Searching</h3>
-            <li><b>Occultation Constraints</b> permits easy searching on
+            <li><b>Occultation/Reflectance Profiles Constraints</b> permits easy searching on
                 the common attributes of both spacecraft-based and ground-based
                 occultation profiles.</li>
```
Should say "occultation or reflectance"

juzen2003 replied:
done

rfrenchseti commented on opus/application/apps/results/test_results.py:
```diff
@@ -404,31 +404,63 @@ def test__get_triggered_tables_voyager(self):
+                    'obs_mission_voyager']
+        self._test_triggered_tables(q, expected)
+
     def test__get_triggered_tables_vgiss_volume(self):
```
After this function, add new tests for the 4 Voyager profile volumes.

juzen2003 replied:
done

rfrenchseti commented on opus/application/apps/results/test_results.py:
```diff
@@ -314,7 +314,7 @@ def test__get_triggered_tables_cocirs(self):
         "[test_results.py] get_triggered_tables: tables triggered by instrument COCIRS"
```
There is a missing test for COCIRS - the test__get_triggered_tables_cocirs_volume. Go ahead and add it after this function. You should probably add two, one for 5xxx and one for 0xxx.

juzen2003 replied:
done

rfrenchseti replied:
Actually, to be consistent with the other tests, I was thinking more of `volumeid=COCIRS_5201` or something rather than adding in `observationtype=`. Like what you did for the VG profiles.

rfrenchseti commented on opus/import/populate_obs_instrument_COCIRS.py:
```diff
@@ -320,3 +785,91 @@ def populate_obs_mission_cassini_COCIRS_sequence_id_OBS(**kwargs):
 ################################################################################
 # THESE ARE SPECIFIC TO OBS_INSTRUMENT_COCIRS
 ################################################################################
+def populate_obs_instrument_coiss_detector_id_OBS(**kwargs):
```
Everything in this section says "coiss" when it should say "cocirs". The function names are wrong in the cirs table schema as well.

juzen2003 replied:
done

rfrenchseti commented on opus/import/table_schemas/obs_instrument_cocirs.json:
```diff
@@ -46,7 +46,9 @@
-            "DETECTOR_ID"
+            "DETECTOR_ID",
+            "FUNCTION",
+            "obs_instrument_coiss_detector_id_<TYPE>"
```
coiss should be cocirs throughout this file

juzen2003 replied:
done

rfrenchseti reviewed (changes requested):
- When importing the entire COCIRS_0xxx and COCIRS_1xxx volumes, there are 18 fatal errors because the volumes before 0402 don't have index files. These volumes should be ignored during import.
- The test suite needs to work both with the test database and when the entire database is loaded. When I import all of COCIRS_0xxx and COCIRS_1xxx, I get the following two failures in `test_search_api.py`:

```
======================================================================
FAIL: [test_search_api.py] /api/stringsearchchoices: volumeid empty
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/seti/src/dave-pds-opus/opus/application/test_api/test_search_api.py", line 498, in test__api_stringsearchchoices_volumeid_empty_bigcache
    self._run_stringsearchchoices_subset(url, expected)
  File "/seti/src/dave-pds-opus/opus/application/test_api/test_search_api.py", line 71, in _run_stringsearchchoices_subset
    self.assertEqual(expected, jdata)
AssertionError: {'choices': ['COISS_2002', 'GO_0017'], 'truncated_results': False, 'reqno': 123} != {'choices': [], 'truncated_results': True, 'reqno': 123}
- {'choices': ['COISS_2002', 'GO_0017'], 'reqno': 123, 'truncated_results': False}
?              -----------------------                                      ^^^^

+ {'choices': [], 'reqno': 123, 'truncated_results': True}
?                                                    ^^^

-------------------- >> begin captured stdout << ---------------------
/__api/stringsearchchoices/volumeid.json?volumeid=&reqno=123
Got:
{'choices': ['COCIRS_0402', 'COCIRS_0403', 'COCIRS_0404', 'COCIRS_0405', 'COCIRS_0406', 'COCIRS_0407', 'COCIRS_0408', 'COCIRS_0409', 'COCIRS_0410', 'COCIRS_0411', 'COCIRS_0412', 'COCIRS_0501', 'COCIRS_0502', 'COCIRS_0503', 'COCIRS_0504', 'COCIRS_0505', 'COCIRS_0506', 'COCIRS_0507', 'COCIRS_0508', 'COCIRS_0509', 'COCIRS_0510', 'COCIRS_0511', 'COCIRS_0512', 'COCIRS_0601', 'COCIRS_0602', 'COCIRS_0603', 'COCIRS_0604', 'COCIRS_0605', 'COCIRS_0606', 'COCIRS_0607', 'COCIRS_0608', 'COCIRS_0609', 'COCIRS_0610', 'COCIRS_0611', 'COCIRS_0612', 'COCIRS_0701', 'COCIRS_0702', 'COCIRS_0703', 'COCIRS_0704', 'COCIRS_0705', 'COCIRS_0706', 'COCIRS_0707', 'COCIRS_0708', 'COCIRS_0709', 'COCIRS_0710', 'COCIRS_0711', 'COCIRS_0712', 'COCIRS_0801', 'COCIRS_0802', 'COCIRS_0803', 'COCIRS_0804', 'COCIRS_0805', 'COCIRS_0806', 'COCIRS_0807', 'COCIRS_0808', 'COCIRS_0809', 'COCIRS_0810', 'COCIRS_0811', 'COCIRS_0812', 'COCIRS_0901', 'COCIRS_0902', 'COCIRS_0903', 'COCIRS_0904', 'COCIRS_0905', 'COCIRS_0906', 'COCIRS_0907', 'COCIRS_0908', 'COCIRS_0909', 'COCIRS_0910', 'COCIRS_0911', 'COCIRS_0912', 'COCIRS_1001', 'COCIRS_1002', 'COCIRS_1003', 'COCIRS_1004', 'COCIRS_1005', 'COCIRS_1006', 'COCIRS_1007', 'COCIRS_1008', 'COCIRS_1009', 'COCIRS_1010', 'COCIRS_1011', 'COCIRS_1012', 'COCIRS_1101', 'COCIRS_1102', 'COCIRS_1103', 'COCIRS_1104', 'COCIRS_1105', 'COCIRS_1106', 'COCIRS_1107', 'COCIRS_1108', 'COCIRS_1109', 'COCIRS_1110', 'COCIRS_1111', 'COCIRS_1112', 'COCIRS_1201', 'COCIRS_1202', 'COCIRS_1203', 'COCIRS_1204', 'COCIRS_1205'], 'truncated_results': True, 'reqno': 123}
Expected:
{'choices': ['COISS_2002', 'GO_0017'], 'truncated_results': False, 'reqno': 123}
Restricted Got:
[]

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: [test_search_api.py] /api/stringsearchchoices: volumeid empty timeout
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/seti/src/dave-pds-opus/opus/application/test_api/test_search_api.py", line 509, in test__api_stringsearchchoices_volumeid_empty_bigcache_timeout
    self._run_stringsearchchoices_subset(url, expected)
  File "/seti/src/dave-pds-opus/opus/application/test_api/test_search_api.py", line 71, in _run_stringsearchchoices_subset
    self.assertEqual(expected, jdata)
AssertionError: {'choices': ['COISS_2002', 'GO_0017'], 'truncated_results': False, 'reqno': 123} != {'choices': [], 'truncated_results': True, 'reqno': 123}
- {'choices': ['COISS_2002', 'GO_0017'], 'reqno': 123, 'truncated_results': False}
?              -----------------------                                      ^^^^

+ {'choices': [], 'reqno': 123, 'truncated_results': True}
?                                                    ^^^

-------------------- >> begin captured stdout << ---------------------
/__api/stringsearchchoices/volumeid.json?volumeid=&reqno=123
Got:
{'choices': ['COCIRS_0402', 'COCIRS_0403', 'COCIRS_0404', 'COCIRS_0405', 'COCIRS_0406', 'COCIRS_0407', 'COCIRS_0408', 'COCIRS_0409', 'COCIRS_0410', 'COCIRS_0411', 'COCIRS_0412', 'COCIRS_0501', 'COCIRS_0502', 'COCIRS_0503', 'COCIRS_0504', 'COCIRS_0505', 'COCIRS_0506', 'COCIRS_0507', 'COCIRS_0508', 'COCIRS_0509', 'COCIRS_0510', 'COCIRS_0511', 'COCIRS_0512', 'COCIRS_0601', 'COCIRS_0602', 'COCIRS_0603', 'COCIRS_0604', 'COCIRS_0605', 'COCIRS_0606', 'COCIRS_0607', 'COCIRS_0608', 'COCIRS_0609', 'COCIRS_0610', 'COCIRS_0611', 'COCIRS_0612', 'COCIRS_0701', 'COCIRS_0702', 'COCIRS_0703', 'COCIRS_0704', 'COCIRS_0705', 'COCIRS_0706', 'COCIRS_0707', 'COCIRS_0708', 'COCIRS_0709', 'COCIRS_0710', 'COCIRS_0711', 'COCIRS_0712', 'COCIRS_0801', 'COCIRS_0802', 'COCIRS_0803', 'COCIRS_0804', 'COCIRS_0805', 'COCIRS_0806', 'COCIRS_0807', 'COCIRS_0808', 'COCIRS_0809', 'COCIRS_0810', 'COCIRS_0811', 'COCIRS_0812', 'COCIRS_0901', 'COCIRS_0902', 'COCIRS_0903', 'COCIRS_0904', 'COCIRS_0905', 'COCIRS_0906', 'COCIRS_0907', 'COCIRS_0908', 'COCIRS_0909', 'COCIRS_0910', 'COCIRS_0911', 'COCIRS_0912', 'COCIRS_1001', 'COCIRS_1002', 'COCIRS_1003', 'COCIRS_1004', 'COCIRS_1005', 'COCIRS_1006', 'COCIRS_1007', 'COCIRS_1008', 'COCIRS_1009', 'COCIRS_1010', 'COCIRS_1011', 'COCIRS_1012', 'COCIRS_1101', 'COCIRS_1102', 'COCIRS_1103', 'COCIRS_1104', 'COCIRS_1105', 'COCIRS_1106', 'COCIRS_1107', 'COCIRS_1108', 'COCIRS_1109', 'COCIRS_1110', 'COCIRS_1111', 'COCIRS_1112', 'COCIRS_1201', 'COCIRS_1202', 'COCIRS_1203', 'COCIRS_1204', 'COCIRS_1205'], 'truncated_results': True, 'reqno': 123}
Expected:
{'choices': ['COISS_2002', 'GO_0017'], 'truncated_results': False, 'reqno': 123}
Restricted Got:
[]

--------------------- >> end captured stdout << ----------------------
```

- (More comments will be coming in future reviews)


Dave: done

rfrenchseti reviewed (changes requested):
- During import, the following WARNING shows up for all COCIRS cube volumes. It shouldn't be there.

```
2021-11-03 20:44:01 | pds.opus_import.main |--| WARNING | Volume "COCIRS_1402" has unexpected body surface geometry files
```


- Dave: done

rfrenchseti reviewed (changes requested):
- The VG_28xx observations don't have preview images. When one of these observations is displayed in the Detail tab, the nav bar at the top displays "(i) Detail <broken icon>undefined"

- Dave: done

rfrenchseti commented on opus/import/table_schemas/internal_def_product_types.json:
```diff
@@ -518,5 +533,325 @@
+    {
+        "pi_dict_context": "OPUS_PRODUCT_TYPE",
+        "pi_dict_name": "vgiss_prof_img_cal",
+        "definition": "Finally calibrated image files (*.TIF) that are geometrically corrected Voyager images. Associated labels (*.LBL) describe the contents of the image files."
```
This definition should be based on the one for other Voyager calibrated GEOMED files, with a small addition for them being part of a profile:

```
"Calibrated and geometrically corrected image files (*_GEOMED.IMG) for Voyager ISS that were used to create a radial scan. Images are in VICAR format and have been calibrated and geometrically corrected at the Ring-Moon Systems Node using the VICAR software package. They are in units of scaled I/F. Associated labels (*.LBL) are text files that contain information about the image. Calibration was performed after apparent flaws in the raw image, including reseau markings and spikes, were removed; these are geometrically corrected versions of the calibrated images (*_CALIB.IMG)."
```

juzen2003 replied:
done

rfrenchseti commented on opus/import/table_schemas/internal_def_product_types.json:
```diff
@@ -518,5 +533,325 @@
+    {
+        "pi_dict_context": "OPUS_PRODUCT_TYPE",
+        "pi_dict_name": "vgiss_prof_img_raw",
+        "definition": "Raw Voyager image files (*.TIF) after decompression. Images are in VICAR2 format. Associated labels (*.LBL) describe the contents of the image files."
```
This should be based on vgiss_raw with a small addition for being part of a radial scan:

```
"Raw image files (*_RAW.IMG) for Voyager ISS that were used to create a radial scan. Images are in VICAR format, uncalibrated, and in units of DN (data number). Associated labels (*.LBL) are text files that contain information about the image. Raw images have not been geometrically corrected and contain significant distortions."
```

juzen2003 replied:
done

rfrenchseti reviewed (changes requested):
- Change the field "Occultation Type" under "Occultation/Reflectance Profile Constraints" to be "Occultation or Profile Type". Update the tooltip to:

"An occultation is a measurement in which a source signal passes through a medium before being observed. Radio means the source is an artificially-emitted radio signal. Solar means the source is the Sun. Stellar means the source is a star other than the Sun. A reflectance profile is a measurement in which a source signal is reflected from a surface and a 1-D slice is recorded."

- Change the tooltip for "Data Quality Score" to:

"A subjective rating of quality of the occultation or reflectance profile."

*** For VG_2810:

- Change Signal Source Name to "N/A"
- Change Spectral Size to "N/A" (not 1)
- While you're at it, fix the other cases in UVIS (both obs and prof) that have a Spectral Size of 1 when Spectral Information Flag is "No"
- Change Shutter Mode of VGISS to "NAONLY" (to agree with Mark's spreadsheet)
- The phase angle (and center phase angle) of a reflectance profile can't possibly be 180

- For all VG28xx: Occultation/Receiver Host Name should be "Voyager 1" or "Voyager 2", not just "Voyager"

Dave: done
    - Note: VG_2810_profile_index.tab is updated to have the correct shutter mode and scan mode values.

rfrenchseti reviewed (changes requested):
- All of the fields marked as "TBD" in Mark's spreadsheet need to be "None" for VG_2801/2/3. There are many fields currently filled in with bad values.

Observed North-Based Incidence (Min)
Observed North-Based Incidence (Max)
Observed North-Based Emission (Min)
Observed North-Based Emission (Max)
Solar Ring Elevation (Min)
Solar Ring Elevation (Max)
Observer Ring Elevation (Min)
Observer Ring Elevation (Max)
Ring Center North-Based Incidence (Min)
Ring Center North-Based Incidence (Max)
Ring Center North-Based Emission (Min)
Ring Center North-Based Emission (Max)
Ring Center Opening Angle to Sun (Min)
Ring Center Opening Angle to Sun (Max)
Ring Center Opening Angle to Observer (Min)
Ring Center Opening Angle to Observer (Max)


- Dave: no change (based on slack discussion), we will keep the mathematic calculations in populate functions for those fields. Add comments in populate_obs_instrument_VGRSS_prof.py to explain how we determine the location of Voyager.

rfrenchseti reviewed (changes requested):
For VG_2803:

- Occultation Type should be RADIO
- Should the Temporal Sampling Interval really be 0?
- The wavelength has the wrong units. It should be something like 13 microns, not 130,000 microns
- Spectrum Size N/A
- Many ring geo parameters should be N/A (see Mark's spreadsheet). Current values are incorrect.

Dave: code is updated, VG_2803 profile index file has been updated to work with the latest version of index files. It will make sure each record in the profile index also has a record exist in the supplemental index

rfrenchseti reviewed (changes requested):
For VG_2802:

- Product type "Vector Tables (ring frame)" has a tooltip that says J2000 frame
- Wavelength and wavelength resolution should not be -99
- Many ring geo entries should be N/A (See Mark's spreadsheet)

Dave: same as above

juzen2003 committed:
Create DESIRED_VOLUMES_IN_VOLSE to store regex patterns used to determine
the desired volumes for import.
- In config_data.py (line 1139-1142), create DESIRED_VOLUMES_IN_VOLSE to
  store regex patterns used to determine which volumes to import.
- In do_import.py (line 596-602), add logic block to check if we want to
  ignore this specific volume in a volset.

juzen2003 committed:
Change "Occultation/Reflectance Profiles Constraints" to "Occultation or
Reflectance Profiles Constraints" in get started guide.

rfrenchseti commented on opus/import/do_import.py:
```diff
@@ -784,10 +796,21 @@ def import_one_index(volume_id, volume_pdsfile, vol_prefix, metadata_paths,
+
+                # This is used to determine if we need to create body surface
+                # geo table when *_summary doesn't exist under metadata.
+                sub_assoc_type = None
```
Add blank line after this assignment.

juzen2003 replied:
done

rfrenchseti commented on opus/import/do_import.py:
```diff
@@ -784,10 +796,21 @@ def import_one_index(volume_id, volume_pdsfile, vol_prefix, metadata_paths,
                 elif 'SUPPLEMENTAL_INDEX' in basename.upper():
                     assoc_type = 'supp_index'
+                    # Check if the index file contains surface geo info, if so,
+                    # we assign 'body_surface_geo' to sub_assoc_dict
```
Actually you assign it to sub_assoc_type, not sub_assoc_dict.

How about a comment like "Check if the supplemental index file contains surface geo info, and it so, mark the sub_assoc_type accordingly."

juzen2003 replied:
done

rfrenchseti commented on opus/import/do_import.py:
```diff
@@ -903,6 +926,29 @@ def import_one_index(volume_id, volume_pdsfile, vol_prefix, metadata_paths,
+
+                    # construct surface geo dict for index files that contains
+                    # surface geo info.
+                    # Format: {opus_id: {target: 'Target_NAME': target}}
```
Shouldn't this be:

{opus_id: {target: {'TARGET_NAME': target}}}

juzen2003 replied:
done

rfrenchseti commented on opus/import/do_import.py:
```diff
@@ -784,10 +796,21 @@ def import_one_index(volume_id, volume_pdsfile, vol_prefix, metadata_paths,
                     assoc_type = 'supp_index'
+                    # Check if the index file contains surface geo info, if so,
+                    # we assign 'body_surface_geo' to sub_assoc_dict
+                    for f in INDEX_FILES_WITH_SURFACE_GEO_INFO:
```
This variable should really be SUPPLEMENTAL_INDEX_FILES_WITH_SURFACE_GEO_INFO

juzen2003 replied:
done

rfrenchseti commented on opus/import/do_import.py:
```diff
@@ -903,6 +926,29 @@ def import_one_index(volume_id, volume_pdsfile, vol_prefix, metadata_paths,
                             key = key.upper()
                             assoc_dict[key] = row
+
+                    # construct surface geo dict for index files that contains
```
"Construct the surface geo dict for supplemental index files that contain surface geo info. The only metadata we put in the surface geo dict is the target name, relying on populate functions to retrieve the necessary metadata later."

juzen2003 replied:
done

rfrenchseti commented on opus/import/config_data.py:
```diff
@@ -1101,3 +1125,10 @@
     'ZET_PUP':              (120.896031417, -40.003147806),
 }
+
+# Index files that contains surface geo info
```
Replace whole comment

"Keywords in supplemental index filenames that indicate they contain surface geo info in the absence of an actual surface geo summary file."

juzen2003 replied:
done

rfrenchseti commented on opus/import/populate_obs_instrument_VGISS_prof.py:
```diff
@@ -0,0 +1,536 @@
+
+# Source: Sun, observer: earth, both of them are at the north side.
+# Incidence angle: the angle between the point where incoming source photons
+# hit the ring, to the north pole of the planet we're looking at (normal vector
```
It's not the north pole, it's whichever pole is on the lit side

rfrenchseti replied:
Also isn't the observer for VGISS the Voyager spacecraft, not Earth? How about this comment to replace the whole thing:

```
# Source: Sun, observer: Voyager. For both of the reflectance profiles,
# the Sun was illuminating the north side of the rings and Voyager was
# observing the rings from the south side. Thus the incidence/emission
# angles and the north-based incidence/emission angles will
# be the same.

# Incidence angle: The angle between the point where the incoming source
# photos hit the ring and the normal to the ring plane on the LIT side of
# the ring. This is always between 0 (parallel to the normal vector) and 90
# (parallel to the ring plane)
```


Dave: done

juzen2003 replied:
done, also update comments for incidence angle on all VG_28xx.

rfrenchseti commented on opus/import/populate_obs_mission_voyager.py:
```diff
@@ -13,6 +13,11 @@
 import impglobals
 import import_util
 
+VG_TARGET_TO_MISSION_PHASE_MAPPING = {
```
Put an underscore in front of the variable name so it stays local to this file

juzen2003 replied:
done, move the variable to config_data.py because it's used in multiple files (VG_28xx)

rfrenchseti commented on opus/import/populate_obs_instrument_VGISS_prof.py:
```diff
@@ -0,0 +1,536 @@
+
+# Opening angle to observer: the angle between the ring surface to the direction
+# where outgoing photons to the observer. Positive if observer is at the north
+# side of the ring , negative if it's at the south side. In this case, observer
```
Actually in this case the observer is on the south side. The proper thing to say is it's always 90 - north-based-emission-angle.

juzen2003 replied:
done

rfrenchseti commented on opus/import/populate_obs_instrument_VGISS_prof.py:
```diff
@@ -0,0 +1,536 @@
+# Ring elevation to observer, same to opening angle except, it's positive if
+# observer is at north side of Jupiter, Saturn, and Neptune, and south side of
+# Uranus. Negative if observer is at south side of Jupiter, Saturn, and Neptune,
+# and north side of Uranus. In this volume, observer is at the north of Saturn,
```
Again the observer is on the south side

juzen2003 replied:
done

rfrenchseti commented on opus/import/populate_obs_instrument_VGISS_prof.py:
```diff
@@ -0,0 +1,536 @@
+    return 90. - min_ea
+
+
+# Opening angle to solar: the angle between the ring surface to the direction
```
Opening angle to Sun

juzen2003 replied:
done, also update in all VG_28xx.

rfrenchseti commented on opus/import/populate_obs_instrument_VGISS_prof.py:
```diff
@@ -0,0 +1,536 @@
+
+    return 90. - inc
+
+# Ring elevation to solar, same to opening angle except, it's positive if
```
Ring elevation to Sun

juzen2003 replied:
done, also update in all VG_28xx.

rfrenchseti commented on opus/import/populate_obs_instrument_VGPPS_prof.py:
```diff
@@ -0,0 +1,581 @@
+# Source SIGMA SGR is at north, observer Voyager is at south. Target: N ring.
+# Source BETA PER is at north, observer Voyager is at south. Target: U ring.
+
+# Incidence angle: the angle between the point where incoming source photons
```
Fix comments to agree with updated comments from VGISS PROF throughout file

juzen2003 replied:
done

rfrenchseti commented on opus/import/populate_obs_instrument_VGPPS_prof.py:
```diff
@@ -0,0 +1,581 @@
+def populate_obs_ring_geometry_VGPPS_emission1_PROF(**kwargs):
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    min_ea = index_row['MAXIMUM_EMISSION_ANGLE']
```
MINIMUM_EMISSION_ANGLE?

juzen2003 replied:
done

rfrenchseti commented on opus/import/populate_obs_instrument_VGUVS_prof.py:
```diff
@@ -0,0 +1,586 @@
+# start time after 1986-01-24T17:10:13.320. Target: N ring.
+# Source IOTA HER is at north, observer Voyager is at south. Target: S ring.
+
+# Incidence angle: the angle between the point where incoming source photons
```
Update comments with changes from VGISS PROF

juzen2003 replied:
done

rfrenchseti commented on opus/import/populate_obs_instrument_VGUVS_prof.py:
```diff
@@ -0,0 +1,586 @@
+    assert abs(cal_inc - inc) <= 0.005, msg
+    return 180. - min_ea
+
+def _is_voyager_at_north(**kwargs):
```
Shouldn't there be some test for the proper datetime here?

juzen2003 replied:
done

rfrenchseti commented on opus/import/populate_obs_instrument_VGUVS_prof.py:
```diff
@@ -0,0 +1,586 @@
+    return (src_name == "DELTA SCO"
+            or (src_name == "SIGMA SGR" and target_name == "U RINGS"))
+
+def _is_voyager_at_north_except_uranus(**kwargs):
```
And also test the datetime here?

juzen2003 replied:
done

rfrenchseti commented on opus/import/populate_obs_instrument_VGRSS_prof.py:
```diff
@@ -0,0 +1,555 @@
+### OBS_PROFILE TABLE ###
+
+def populate_obs_occultation_VGRSS_occ_type_PROF(**kwargs):
+    return 'STE' # There are no SUN occultations
```
RSS is Radio

juzen2003 replied:
done

rfrenchseti commented on opus/import/populate_obs_instrument_VGRSS_prof.py:
```diff
@@ -0,0 +1,555 @@
+    return src_name1
+
+def populate_obs_occultation_VGRSS_host_PROF(**kwargs):
+    return 'voyager'
```
Should be "Voyager 1" or "Voyager 2"

juzen2003 replied:
done, also update on all VG_28xx

rfrenchseti commented on opus/import/populate_obs_instrument_VGRSS_prof.py:
```diff
@@ -0,0 +1,555 @@
+    return 180.
+
+# Source: Voyager RSS is at south, observer: earth is at north.
+# Incidence angle: the angle between the point where incoming source photons
```
Update comment from VGISS changes

juzen2003 replied:
done

rfrenchseti commented on opus/import/populate_obs_instrument_VGRSS_prof.py:
```diff
@@ -0,0 +1,555 @@
+    index_row = metadata['index_row']
+
+    target_name = populate_target_name_from_index(**kwargs)
+    if target_name =='U RINGS':
```
Perhaps break out all the "U RINGS" into another function like you did for the other instruments? Something like `_is_voyager_at_uranus`. Also add a space after `==`.

juzen2003 replied:
done

rfrenchseti commented on opus/import/populate_obs_instrument_VGRSS_prof.py:
```diff
@@ -0,0 +1,555 @@
+# ring, to the direction where outgoing photons to the observer. 0-90 when
+# observer is at the north side of the ring, and 90-180 when it's at the south
+# side.
+# Since observer is at north, north based emission angle is between 0-90.
```
Isn't the observer at the south?

juzen2003 replied:
Based on slack discussion, the observer is at north.

rfrenchseti commented on opus/import/populate_obs_instrument_VGRSS_prof.py:
```diff
@@ -0,0 +1,555 @@
+# side of the ring , negative if it's at the south side. In this case, observer
+# is at the north side, so it's ea - 90 (or 90 - inc). For reference, if
+# observer is at the south side, then oa is 90 - ea.
+# Observer is at north, so it's ea - 90 (positive 0-90)
```
Isn't the observer at the south?

juzen2003 replied:
same as the above.

rfrenchseti commented on opus/import/populate_obs_instrument_VGRSS_prof.py:
```diff
@@ -0,0 +1,555 @@
+    target_name = populate_target_name_from_index(**kwargs)
+    if target_name =='U RINGS':
+        max_ea = index_row['MAXIMUM_EMISSION_ANGLE']
+        el = - (max_ea - 90.) # negative
```
Might be easier to read as 90 - max

juzen2003 replied:
done

rfrenchseti commented on opus/import/populate_obs_instrument_VGRSS_prof.py:
```diff
@@ -0,0 +1,555 @@
+# side of the ring, negative if it's at the south side. In this case, source
+# is at the south side, so it's - (90 - inc). For reference, if source is at the
+# north side, then oa is 90 - inc.
+# Source is at south of the ring, so it's neagtive which is inc - 90
```
Isn't the observer at the south?

juzen2003 replied:
Based on slack discussion, the observer is at north.

rfrenchseti commented on opus/import/populate_obs_instrument_VGRSS_prof.py:
```diff
@@ -0,0 +1,555 @@
+    target_name = populate_target_name_from_index(**kwargs)
+    if target_name =='U RINGS':
+        inc = populate_obs_ring_geometry_VGRSS_incidence2_PROF(**kwargs)
+        el = - (inc - 90.) # positive
```
Might be easier to read as 90-inc

juzen2003 replied:
done

juzen2003 committed:
Add triggered table tests for VG_2810/2801/2802/2803.
- In test_results.py (line 470-513), add test__get_triggered_tables_*
  for VG_2810/2801/2802/2803.

juzen2003 committed:
Add triggered tables tests for COCIRS_5xxx & COCIRS_0xxx.
- In test_results.py (line 324-345), add test__get_triggered_tables_cocirs_volume_5xxx
  and test__get_triggered_tables_cocirs_volume_0xxx

juzen2003 committed:
Replace _coiss_ to _cocirs_ in both populate_obs_instrument_COCIRS.py and
obs_instrument_cocirs.json.

juzen2003 committed:
Put COCIRS_0xxx and COCIRS_1xxx in VOLSETS_WITH_SURFACE_GEO since they
will have body surface geo imported.
- In config_data.py (line 44,45), add COCIRS_0xxx and COCIRS_1xxx in
  VOLSETS_WITH_SURFACE_GEO.

juzen2003 committed:
Make sure navbar detail (i) has the a proper image even if preview image
is not available by setting ignore_missing to False in get_pds_preview_images
function. (line 815 in results/views.py).

juzen2003 committed:
- Update the followings based on Rob's feedback:
- Change "Occultation Type" under "Occultation/Reflectance Profile
  Constraints" to be "Occultation or Profile Type" and update the tooltip.
  (obs_profile.json line 63-73)
- Change tooltip for "Data Quality Score" (obs_profile.json line 178)
- For VG_2810:
    - Update signal source name to "N/A" (populate_obs_instrument_VGISS_prof.py
      line 237)
    - Update spectral size to "N/A" (populate_obs_instrument_VGISS_prof.py
      line 191)
    - Fixed the spectral size to "N/A" in COUIVS (populate_obs_instrument_COUVIS_prof.py
      line 196)
    - Update phase angle to read from profile index file (populate_obs_instrument_VGISS_prof.py
      line 303-315)
    - Update the profile index file to have the correct values for
      shutter mode and scan mode.
- Update receiver host name to read from profile index file for VG_28xx.

juzen2003 committed:
Fixed the spec_size for COUVIS. When spec_size is 1, spec_flag will be
Yes. When spec_size is less than 1 or not available, spec_flag will be No.

juzen2003 committed:
Update the definition of "vgiss_prof_img_raw" and "vgiss_prof_img_cal"
in internal_def_product_types.json (line 850, 855)

juzen2003 committed:
Update comments to include the method of how we determine the location
of Voyager. (populate_obs_instrument_VGRSS_prof.py, line 326-329)

juzen2003 committed:
Update to fetch wavelength/temporal sampling interval value from
supplemental index files in VG_28xx.
- In VG_2803: (populate_obs_instrument_VGRSS_prof.py)
    - Update the occultation type to radio (line 199).
    - Update spectrum size to None (line 190)
    - Update to fetch wavelength and temporal sampling interval from
      supplemental index (line 129,135,223)
- In all populate_obs_instrument_VG*_prof.py, Update to fetch wavelength
  and temporal sampling interval from supplemental index.
- In internal_def_product_types.json (line 725), change the tooltip from
  J2000 frame to ring frame.

juzen2003 committed:
- Update comments in import_one_index function in do_import.py (line
817-819, 940-945)
- Rename "INDEX_FILES_WITH_SURFACE_GEO_INFO" to
  "SUPPLEMENTAL_INDEX_FILES_WITH_SURFACE_GEO_INFO".

juzen2003 committed:
Move VG_TARGET_TO_MISSION_PHASE_MAPPING to config_data and update the
comments of incidence angle in all populate_obs_instrument_*_prof.py files.

juzen2003 committed:
Update comments on opening angle and ring elevation for all populate_obs_instrument_VG*_prof.py

juzen2003 committed:
Update _is_voyager_at_north & _is_voyager_at_north_except_uranus by adding
test for datetime.
- In config_data.py (line 1150-1154), add THRESHOLD_START_TIME_VG_AT_NORTH
  as the threshold time to determine whether Voyager is at North or South.
- In populate_obs_instrument_VGPPS_prof.py (line 382-389, 400-406), add
  test for datetime in _is_voyager_at_north* functions.
- In populate_obs_instrument_VGUVS_prof.py (line 384-391, 402-408), add
  test for datetime in _is_voyager_at_north* functions.

juzen2003 committed:
Add _is_voyager_at_uranus to determine if the Voyager is targeting U RINGS.
- In populate_obs_instrument_VGRSS_prof.py (line 562-567), create
  _is_voyager_at_uranus to determine if the Voyager is targeting U RINGS.
- Replace all "if target_name =='U RINGS':" by _is_voyager_at_uranus calls
  in populate_obs_instrument_VGRSS_prof.py.

juzen2003 committed:
Re-arrange mathematic calculations of ring geometry for VG_28xx.

juzen2003 committed:
Add datatime tests for _is_voyager_at_north* functions for VGPPS & VGUVS.
- In populate_obs_instrument_VGPPS_prof.py (line 378-418), add datetime
  tests for for _is_voyager_at_north* functions.
- In populate_obs_instrument_VGUVS_prof.py (line 380-420), add datetime
  tests for for _is_voyager_at_north* functions.

juzen2003 committed:
Update to fetch wavelength data from profile index for VG_2803 because
values in supplemental index files are not correct.

juzen2003 committed:
Organize the code to have populate_obs_instrument_COCIRS_cube.py import
COCRIS_0xxx/COCIRS_1xxx, and populate_obs_instrument_COCIRS_obs.py import
COCIRS_5xxx/COCIRS_6xxx. Also update test results.

juzen2003 committed:
Fixed test__api_stringsearchchoices_volumeid_empty_bigcache* by increasing
limit to 400 so that COISS_2002 & GO_0017 can be returned.
- In test_search_api.py (line 494, 505), add limit=400 in api url so that
  COISS_2002 & GO_0017 can be returned when running the test against full
  database.

juzen2003 committed:
Move 'VG_TARGET_TO_MISSION_PHASE_MAPPING' from config_data.py to
populate_obs_mission_voyager.py.

rfrenchseti commented on opus/application/test_api/responses/results_vg_rss_2_u_occ_1986_024_s43_four_i_metadata.json:
```diff
@@ -0,0 +1,141 @@
+        "occdepth2": null,
+        "occwlband": "S",
+        "occsource": "Voyager 2",
+        "occreceiverhost": "Dsn-43",
```
The receiver host needs to be in the same format as what's already in OPUS for Cassini RSS. "DSN 43 (Canberra 70m)". See the Cassini RSS code for the appropriate mapping.

rfrenchseti commented on opus/import/populate_obs_instrument_VGPPS_prof.py:
```diff
@@ -0,0 +1,622 @@
+    msg = (f"{index_row['FILE_SPECIFICATION_NAME']} start time" +
+           f"{index_row['START_TIME']} and Voyager location do not match.")
+
+    isAtNorth = (src_name == 'DELTA SCO'
```
Wrong format for a variable name

juzen2003 committed:
- Fixed the receiver host for VGRSS by utilizing the DSN_NAMES mapping from
CORSS.
- Rename 'isAtNorth' to 'is_at_north'
- Update test results by calling the specific volume, for example:
  test__get_triggered_tables_cocirs_volume_5201
  test__get_triggered_tables_cocirs_volume_0406

juzen2003 committed:
Removing warning messages about missing surface geo info for COCIRS_0xxx/1xxx.
- In test_search_api.py (line 499), remove test__api_stringsearchchoices_volumeid_empty_bigcache_timeout.
- In populate_obs_instrument_VGRSS_prof.py (line 255), convert dsn to
  integer so that it can be properly looked up in DSN_NAMES dictionary.
- In config_data.py (line 44-45), update VOLSETS_WITH_SURFACE_GEO to
  COCIRS_0xxx_CUBE_EQUI and COCIRS_1xxx_CUBE_EQUI since they both contain
  surface geo info.
- In do_import.py (line 748-753), use volset_info to check against
  VOLSETS_WITH_RING_GEO and VOLSETS_WITH_SURFACE_GEO.

juzen2003 committed:
Update test results for COCIRS_0xxx/1xxx and VG_2803.

juzen2003 committed:
Update test__api_stringsearchchoices_volumeid_empty_bigcache to have limit=10000

juzen2003 committed:
Remove unwanted comments in config_data.py and test cases in import_for_tests.sh.

juzen2003 committed:
Update VGRSS test results after using files under KM00_1 as the primary
file spec for VG_2803.

juzen2003 committed:
Remove unwanted volumes (COCIRS_1002,COCIRS_0010) in import_for_tests.sh

juzen2003 committed:
Update spec_size to be None for VG_2801, and also change the return of
populate_obs_occultation_VGISS_source_PROF to None.

juzen2003 committed:
Fixed test results of VG_2801 & VG_2810 after updating the source name
and spec size for them.

juzen2003 committed:
Update the definition of cocirs_cube & cocirs_extra in internal_def_product_types.json. (line 345, 350)

juzen2003 committed:
Update test results (files) for VG_28xx after shelves/info/metadata are
updated.
